### PR TITLE
Install tweaks

### LINKED ISSUE #2670: Unable to clean update on Ubuntu with --skip-smudge enabled
On Ubuntu if git-lfs is installed with '--skip-smudge --system' updating with 'apt' results in errors.  I have to run 'git lfs install --system' and then rerun the update.

`carlg@carlg:~$` sudo git lfs install --skip-smudge --system
Git LFS initialized.
carlg@carlg:~$ sudo git lfs install --skip-smudge
Git LFS initialized.
carlg@carlg:~$ git lfs env
git-lfs/2.3.1 (GitHub; linux amd64; go 1.8.1)
git version 2.14.1

LocalWorkingDir=
LocalGitDir=
LocalGitStorageDir=
LocalMediaDir=
LocalReferenceDir=
TempDir=/tmp/git-lfs
ConcurrentTransfers=3
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
LfsStorageDir=lfs
AccessDownload=none
AccessUpload=none
DownloadTransfers=basic
UploadTransfers=basic
git config filter.lfs.process = "git-lfs filter-process --skip"
git config filter.lfs.smudge = "git-lfs smudge --skip -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"

carlg@carlg:~$ apt list --installed | grep 'git-'

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

git-core/now 1:2.14.1-1.1~ppa0~ubuntu16.04.1 all [installed,upgradable to: 1:2.14.2-1~ppa0~ubuntu16.04.1]
git-lfs/xenial,now 2.3.1 amd64 [installed,upgradable to: 2.3.3]
git-man/now 1:2.14.1-1.1~ppa0~ubuntu16.04.1 all [installed,upgradable to: 1:2.14.2-1~ppa0~ubuntu16.04.1]
git-svn/now 1:2.14.1-1.1~ppa0~ubuntu16.04.1 all [installed,upgradable to: 1:2.14.2-1~ppa0~ubuntu16.04.1]

carlg@carlg:~$ sudo apt install git-lfs
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages were automatically installed and are no longer required:
  linux-headers-4.4.0-93 linux-headers-4.4.0-93-generic linux-image-4.4.0-93-generic linux-image-extra-4.4.0-93-generic
Use 'sudo apt autoremove' to remove them.
The following packages will be upgraded:
  git-lfs
1 upgraded, 0 newly installed, 0 to remove and 89 not upgraded.
Need to get 2,813 kB of archives.
After this operation, 1,024 B of additional disk space will be used.
Get:1 https://packagecloud.io/github/git-lfs/ubuntu xenial/main amd64 git-lfs amd64 2.3.3 [2,813 kB]
Fetched 2,813 kB in 1s (1,466 kB/s)
(Reading database ... 335541 files and directories currently installed.)
Preparing to unpack .../git-lfs_2.3.3_amd64.deb ...
Global Git LFS configuration has been removed.
Unpacking git-lfs (2.3.3) over (2.3.1) ...
Processing triggers for man-db (2.7.5-1) ...
Setting up git-lfs (2.3.3) ...
The filter.lfs.smudge attribute should be "git-lfs smudge -- %f" but is "git-lfs smudge --skip -- %f"
Run `git lfs install --force` to reset git config.
dpkg: error processing package git-lfs (--configure):
 subprocess installed post-installation script returned error exit status 2
E: Sub-process /usr/bin/dpkg returned an error code (1)

carlg@carlg:~$ cat /etc/issue
Ubuntu 16.04.3 LTS \n \l

carlg@carlg:~$ git lfs install
Git LFS initialized.

carlg@carlg:~$ sudo git lfs install --system
The filter.lfs.smudge attribute should be "git-lfs smudge -- %f" but is "git-lfs smudge --skip -- %f"
Run `git lfs install --force` to reset git config.
carlg@carlg:~$ sudo git lfs install --system --force
Git LFS initialized.

carlg@carlg:~$ sudo apt install git-lfs
Reading package lists... Done
Building dependency tree       
Reading state information... Done
git-lfs is already the newest version (2.3.3).
The following packages were automatically installed and are no longer required:
  linux-headers-4.4.0-93 linux-headers-4.4.0-93-generic linux-image-4.4.0-93-generic linux-image-extra-4.4.0-93-generic
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 89 not upgraded.
1 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Do you want to continue? [Y/n] y
Setting up git-lfs (2.3.3) ...
Git LFS initialized.`

### PR DESCRIPTION
**Pull request by technoweenie**
Fixes #2670. This PR fixes two common, but not fatal, problems that can arise when running `git lfs install`:

1. The values from `git lfs install --skip-smudge` can not be upgraded with `git lfs install`
2. An unexpected config value should not be overwritten. Instead, print a warning _without changing the exit code_. This way installations don't fail, but a user is told if their config values are unexpected.

### CODE REVIEW

**Comment by adambet3** [comment]
Ok

**Comment by angkanasoisom** [comment]
#2670

File: commands/command_install.go [hunk 1]
```diff
@@ -25,8 +25,9 @@ func installCommand(cmd *cobra.Command, args []string) {
 	}
 
 	if err := lfs.InstallFilters(opt, skipSmudgeInstall); err != nil {
-		Error(err.Error())
-		Exit("Run `git lfs install --force` to reset git config.")
+		Print("WARNING: %s", err.Error())
```
Comment (Igorcazakov) [inline commands/command_install.go]: [.pdf](https://github.com/git-lfs/git-lfs/files/1692623/default.pdf)

File: lfs/setup.go [hunk 2]
```diff
@@ -28,6 +28,22 @@ var (
 		postMergeHook,
 	}
 
+	upgradeables = map[string][]string{
```
Comment (Igorcazakov) [inline lfs/setup.go]: String

**Pull request merged**